src/UCTNode.cpp

```diff
@@ -265,7 +272,9 @@ UCTNode* UCTNode::uct_select_child(int color, bool is_root) {
         fpu_reduction = cfg_fpu_reduction * std::sqrt(total_visited_policy);
     }
     // Estimated eval for unknown nodes = original parent NN eval - reduction
-    auto fpu_eval = get_net_eval(color) - fpu_reduction;
+    auto visits = get_visits();
+    auto parent_eval = (cfg_dyn_fpu && visits > 0) ? get_raw_eval(color) : get_net_eval(color);
```

Comment:
These dyn_fpu changes seems unrelated to dynamic komi changes?

Reply:
With --pos or --neg it's also impossible to use net_eval for the fpu of a child, since the parent and the child have different sides to move, so I opt to use dynamic eval.

Reply:
Might be worth testing it in regular Leela then. If it's a wash I'd take the change to make your merge easier.